packages/grid/_modules_/grid/models/gridOptions.tsx

```diff
@@ -225,33 +226,40 @@ export interface GridOptions {
   /**
    * Callback fired when a click event comes from a cell element.
    * @param param With all properties from [[GridCellParams]].
+   * @param event [[React.MouseEvent]].
    */
-  onCellClick?: (param: GridCellParams) => void;
+  onCellClick?: (param: GridCellParams, event: React.MouseEvent) => void;
   /**
    * Callback fired when a double click event comes from a cell element.
    * @param param With all properties from [[CellParams]].
    */
-  onCellDoubleClick?: (param: GridCellParams) => void;
+  onCellDoubleClick?: (param: GridCellParams, event: React.MouseEvent) => void;
   /**
    * Callback fired when a hover event comes from a cell element.
    * @param param With all properties from [[GridCellParams]].
    */
-  onCellHover?: (param: GridCellParams) => void;
+  onCellOver?: (param: GridCellParams, event: React.MouseEvent) => void;
+  onCellOut?: (param: GridCellParams, event: React.MouseEvent) => void;
+  onCellEnter?: (param: GridCellParams, event: React.MouseEvent) => void;
+  onCellLeave?: (param: GridCellParams, event: React.MouseEvent) => void;
   /**
    * Callback fired when a click event comes from a row container element.
    * @param param With all properties from [[GridRowParams]].
    */
-  onRowClick?: (param: GridRowParams) => void;
+  onRowClick?: (param: GridRowParams, event: React.MouseEvent) => void;
   /**
    * Callback fired when a click event comes from a row container element.
    * @param param With all properties from [[RowParams]].
    */
-  onRowDoubleClick?: (param: GridRowParams) => void;
+  onRowDoubleClick?: (param: GridRowParams, event: React.MouseEvent) => void;
```

Comment:
If we expose the event, it should be the first arg, per convention. I don't think that any other approach is an option.

```suggestion
  onRowDoubleClick?: (event: React.MouseEvent, param: GridRowParams) => void;
```

Reply:
if we swap them we add a breaking change

Reply:
We have to consider that some events won't have the `event` obj ie `onRowSelected`

Reply:
Why `onRowSelected` doesn't have an event?

Reply:
But we can give this approach a try. I wouldn't be surprised if we get developers to ask why it's inconsistent. I think that as long as we are in v4-alpha, we can experiment with it 👍

Reply:
> Why `onRowSelected` doesn't have an event? It can either be triggered from a click or keyboard event. Maybe a developer would only want to handle the keyboard event? (checking event.type) 🤷‍♂️

Well it's a grid event not a native DOM event, hence why it should not have the DOM event. 
OnClick can do other things than selecting a row. Selecting a row is just an action not tight to the user interaction such as a click.

If the user wants to handle the event then he should hook into the dom events such onKeyDown or onClick directly.

Reply:
I don't share this perspective because +90% of the events of the gird originate from DOM events. The other rare cases are when responding to timeout, network, or prop updates. But fine, we will learn more by doing things differently and paying close attention to what the feedback from the ~users~ developers are 👍 , for now, the switching cost is low, so better leverage it.